scripts/bootloader/keygen.py

```diff
@@ -8,28 +8,35 @@
 from ecdsa import SigningKey
 from ecdsa.curves import NIST256p
 import argparse
+import sys
 
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(
         description="Generate PEM file.",
         formatter_class=argparse.RawDescriptionHelpFormatter)
-    parser.add_argument("--output-private", required=False,
-                        help="Output private key file name.")
-    parser.add_argument("--output-public", required=False,
-                        help="Output public key file name.")
+    parser.add_argument("--private", required=False, action="count",
+                        help="Output a private key.")
+    parser.add_argument("--public", required=False, action="count",
+                        help="Output a public key.")
+    parser.add_argument("--out", "-out", "-o", required=False,
+                        type=argparse.FileType('wb'), default=sys.stdout.buffer,
+                        help="Output to specified file instead of stdout.")
+    parser.add_argument("--in", "-in", "-i", required=False, dest="infile",
+                        type=argparse.FileType('rb'),
+                        help="Read private key from specified PEM file instead of generating it.")
+
     args = parser.parse_args()
 
-    sk = SigningKey.generate(curve=NIST256p)
-    if args.output_private:
-        with open(args.output_private, 'wb') as sk_file:
-            sk_file.write(sk.to_pem())
+    sk = SigningKey.from_pem(args.infile.read()) if args.infile else SigningKey.generate(curve=NIST256p)
+
+    if args.private:
+        args.out.write(sk.to_pem())
 
-    if args.output_public:
+    if args.public:
         vk = sk.get_verifying_key()
         vk.to_pem()
-        with open(args.output_public, 'wb') as vk_file:
-            vk_file.write(vk.to_pem())
+        args.out.write(vk.to_pem())
```

Comment:
What happens if you give both `--private` and `--public` arguments?

Reply:
Good point, will fix or explain.

Reply:
Fixed (mutually_exclusive_group)